saleor/order/actions.py

```diff
@@ -607,32 +594,318 @@ def create_refund_fulfillment(
     refunded line. The refund fulfillment can have assigned lines with the same
     products but with the different stocks.
     """
+
+    _process_refund(
+        requester=requester,
+        order=order,
+        payment=payment,
+        order_lines_to_refund=order_lines_to_refund,
+        fulfillment_lines_to_refund=fulfillment_lines_to_refund,
+        amount=amount,
+        refund_shipping_costs=refund_shipping_costs,
+    )
+
     with transaction.atomic():
         refunded_fulfillment, _ = Fulfillment.objects.get_or_create(
             status=FulfillmentStatus.REFUNDED, order=order
         )
         already_refunded_lines = list(refunded_fulfillment.lines.all())
-        all_refunded_lines: Dict[int, Tuple[int, OrderLine]] = dict()
-        refund_order_lines = _refund_fulfillment_for_order_lines(
-            order_lines_to_refund=order_lines_to_refund,
-            order_lines_quantities_to_refund=order_lines_quantities_to_refund,
-            already_refunded_lines=already_refunded_lines,
-            refunded_fulfillment=refunded_fulfillment,
-            all_refunded_lines=all_refunded_lines,
+        _move_order_lines_to_target_fulfillment(
+            order_lines_to_move=order_lines_to_refund,
+            lines_in_target_fulfillment=already_refunded_lines,
+            target_fulfillment=refunded_fulfillment,
+        )
+        _move_fulfillment_lines_to_target_fulfillment(
+            fulfillment_lines_to_move=fulfillment_lines_to_refund,
+            lines_in_target_fulfillment=already_refunded_lines,
+            target_fulfillment=refunded_fulfillment,
+        )
+
+        Fulfillment.objects.filter(order=order, lines=None).delete()
+
+    return refunded_fulfillment
+
+
+def create_replace_order(
+    requester: Optional["User"],
+    original_order: "Order",
+    order_lines_to_replace: List[OrderLineData],
+    fulfillment_lines_to_replace: List[FulfillmentLineData],
+) -> "Order":
+    """Create draft order with lines to replace."""
+    # fetch original order to not lose reference for original_order
+    replace_order = Order.objects.get(pk=original_order.pk)
+    replace_order.pk = None
+    replace_order.token = None
+    replace_order.status = OrderStatus.DRAFT
+    replace_order.created = now()
+    replace_order.save()
```

Comment:
I'm not sure that we should copy the order in this way. 
Currently, we copy all payments, gift cards, shipping method, order weight, and total price.

Reply:
What about the situation the new order weight is too low to send it with the original shipping method?

Reply:
- payment - Payment object stores the relation for order. The new order will not have a relation to the payment from original_order
- giftcards - same thing - ManyToMany relations are stored in separate obejct
- shipping-method, order weight it should as now
- voucher- this relations should be cleaned
-  good catch with total_price and weights. I will re-calculate them